ds701_book/07-Clustering-II-in-practice.qmd

```diff
@@ -324,44 +611,48 @@ The second question we face in evaluating a clustering is how many clusters are
 
 In practice, to use $k$-means or most other clustering methods, one must choose $k$, the number of clusters, via some process.
 
-### Inspecting Clustering Error
+## Inspecting Clustering Error
 
 The first thing you might do is to look at the $k$-means objective function  and see if it levels off after a certain point.
 
 Recall that the $k$-means objective can be considered the clustering "error".
 
 If the error stops going down, that would suggest that the clustering is not improving as the number of clusters is increased.
 
+Let's calculate the error for 1-11 clusters.
+
 ```{python}
-#| slideshow: {slide_type: fragment}
-#| tags: [hide-input]
+#| code-fold: false
 error = np.zeros(11)
 for k in range(1,11):
     kmeans = KMeans(init='k-means++', n_clusters = k, n_init = 10)
     kmeans.fit_predict(X)
     error[k] = kmeans.inertia_
 ```
 
+::: {.content-visible when-profile="slides"}
+## Inspecting Clustering Error
+:::
+
 For our synthetic data, here is the $k$-means objective, as a function of $k$:
 
 ```{python}
-#| hide_input: true
-#| slideshow: {slide_type: fragment}
-#| tags: [hide-input]
 plt.plot(range(1, len(error)), error[1:], 'o-')
 plt.xlabel('Number of clusters')
 plt.title(r'$k$-means clustering performance of synthetic data')
 plt.ylabel('Error');
 ```
 
-__Warning__: This synthetic data is not at all typical.   You will almost never see such a sharp change in the error function as we see here.
+::: {.callout-warning}
+This synthetic data is not at all typical.   You will almost never see such a sharp change in the error function as we see here.
```

Comment:
We can remove the space after the period here.

Reply:
Done.